lib/light-service/orchestrator.rb

```diff
@@ -0,0 +1,79 @@
+module LightService
+  module Orchestrator
+    def self.extended(base_class)
+      base_class.extend ClassMethods
+    end
+
+    module ClassMethods
+      def with(data = {})
+        @context = LightService::Context.make(data)
+        self
+      end
+
+      def reduce(steps, context = @context)
+        steps.each_with_object(context) do |step, ctx|
+          if step.respond_to?(:execute)
+            step.execute(ctx)
+          elsif step.respond_to?(:call)
+            step.call(ctx)
+          else
+            raise 'Pass either an action or organizer'
+          end
+        end
+      end
+
+      def reduce_until(condition_block, steps)
+        lambda do |ctx|
+          loop do
+            ctx = scoped_reduction(ctx, steps)
+            break if condition_block.call(ctx) || ctx.failure?
+          end
+
+          ctx
+        end
+      end
+
+      def reduce_if(condition_block, steps)
+        lambda do |ctx|
+          ctx = scoped_reduction(ctx, steps) if condition_block.call(ctx)
+          ctx
+        end
+      end
+
+      def execute(code_block)
+        lambda do |ctx|
+          ctx = code_block.call(ctx)
+          ctx
+        end
+      end
+
+      def iterate(collection_key, steps)
+        lambda do |ctx|
+          collection = ctx[collection_key]
+          item_key = collection_key.to_s.singularize.to_sym
```

Comment:
No, unfortunately. ActiveSupport was added when we introduced I18n message lookups for error or success messages.  I spiked out the removal of AS, and it turned out to be too big of an effort for now. But eventually, I'd like to see no Gem dependency in the project.